Add some tweaks to the Makefile to make it behave more like a Makefile

mjaschen opened the pull request:
- use *make* variables
- remove empty target file in case of an error (the target file is
  created by the shell redirection before `mandoc` runs)

mjaschen committed:
Add some tweaks to the Makefile to make it behave more like a Makefile ;-)

- use Make variables
- remove empty target file in case of an error (the target file is
  created by the shell redirection before `mandoc` runs)

hukl commented:
Kannst du ggf. noch mit Kommentaren annotieren was die make Variablen genau tun? Kenne mich da nicht so aus und so direkt erschließt sich mir die vermeintliche Verbesserung nicht.

mjaschen committed:
add some comments to the Makefile

- `src` und `tgt` habe ich entfernt, da sie nicht notwendig sind -- make
  hat für die Stellen, an denen sie verwendet wurden eigene Variablen:
  - $@ - aktuelles Target
  - $< - erstes Prerequisite (hier: hostmux.mandoc)
- das Starten einer Sub-Shell für `dirname` habe ich entfernt und mit
  der make-eigenen Funktion `dir` ersetzt (ist portabler)
- die Notwendigkeit des Konstrukts zur Fehlerbehandlung während der
  `mandoc`-Ausführung habe ich im Makefile mit einem Comment erklärt

mjaschen commented:
Habe im Commit-Text sowie direkt im Makefile Kommentare hinterlassen. Hoffe, das passt dir so.
